### PR TITLE
feat(client): make `correlationKey` step optional

### LINKED ISSUE #13649: Java Client: Make correlationKey step optional
**Is your feature request related to a problem? Please describe.**
Currently, when you want to start a process instance with a message, you need to provide a correlationKey, even though it's not [needed](https://docs.camunda.io/docs/components/modeler/bpmn/message-events/#message-start-events).  

Customer anecdote:
"We have to set a value we don´t need, so maybe a confused developer who will search for the correlationKey if he/she don´t know this issue."

```java
final ZeebeFuture<PublishMessageResponse> send = this.client.newPublishMessageCommand()
.messageName("MessageRestartLasttest")
.correlationKey("walter")
.variables(Map.of("anzahl", anzahlLoop))
.send();
``` 

**Describe the solution you'd like**
It should be possible to skip PublishMessageCommandStep2. It is up to the coder to know if and when a correlationKey is required by the given BPMN message event.

**Describe alternatives you've considered**
It can be set to "" (empty string) if you do not need it for that purpose, e.g.:
```java
final PublishMessageResponse publishMessageResponse = client.newPublishMessageCommand()
                    .messageName("start-process")
                    .correlationKey("")
                    .variables(Map.of("variable1", "value1"))
                    .send()
                    .join();
``` 

**Additional context**
https://jira.camunda.com/browse/SUPPORT-17405

### PR DESCRIPTION
**Pull request by ce-dmelnych**
## Description

This PR adds the `withoutCorrelationKey` method to the `PublishMessageCommandStep1` interface, allowing optional use of `correlationKey` in message publishing. 
This update caters to scenarios where a correlationKey is not necessary, particularly in cases involving message start events.

With the new `withoutCorrelationKey` method, it is now possible to seamlessly progress to `PublishMessageCommandStep3` without the need to explicitly define a `correlationKey`. 

## Related issues

closes #13649 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java [hunk 1]
```diff
@@ -41,6 +41,18 @@ interface PublishMessageCommandStep2 {
...
+     * Skip specifying a correlation key for the message.
+     *
+     * <p>This method allows the message to be published without a correlation key, making it
+     * suitable for scenarios where the correlation key is not necessary (e.g. for the message start
+     * event). When used, this will create a new process instance without checking for an active
+     * instance with the same correlation key.
+     *
+     * @return the builder for this command, continuing to the next step without a correlation key.
+     */
+    PublishMessageCommandStep3 skipCorrelationKey();
```
Comment (korthout) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java:55]: For users, the `skip` wording might not be clear. It describes the act of skipping specifying the correlation key, but it needs to fit better from the perspective of building a publish message command. What does this mean for the message that is published? See example below:

```java
client
  .newPublishMessageCommand()
  .messageName("my_message")
  .skipCorrelationKey()
  .send()
  .join();
```

🔧 Instead, we could name this `noCorrelationKey()`. That would signal to users what the method is used for. They will publish a message with no correlation key. They'd choose between `correlationKey(String)` and `noCorrelationKey()`.

```java
client
  .newPublishMessageCommand()
  .messageName("my_message")
  .noCorrelationKey()
  .send()
  .join();
```

🔧 Another alternative is `withoutCorrelationKey()`. This is my favorite and was [suggested by someone else](https://github.com/camunda/zeebe/issues/13649#issuecomment-1826053751).

```java
client
  .newPublishMessageCommand()
  .messageName("my_message")
  .withoutCorrelationKey()
  .send()
  .join();
```

**Review by korthout — APPROVED** [review]
Nice work @ce-dmelnych 🎉 And sorry for the delay in reviewing 🙇 

👍 I really like the Javadoc on the interface

🔧 I have one suggestion for improvement. Please have a look before merging.

🔧 Before merging, also please rebase your changes into one or more commits with meaningful messages. The team typically chooses to rebase their changes on top of `main` rather than merging upstream changes with merge commits. This avoids unnecessary commits like `Merge branch 'main' into ...` from cluttering the repository history.

File: clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java (hunk 1 above)
Comment (lzgabel) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java:55]: `withoutCorrelationKey` +1

Comment (ce-dmelnych) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java]: Hi @korthout ! 
Thank you for your detailed and informative review! 🙇

I'm glad to hear that you liked the Javadoc on the interface. 😊

I've implemented your suggestion to improve the naming of the new method and have already rebased my changes into a single commit in my feature branch.
Thank you for guiding me in the right direction!

Could you please let me know if there are any other specific changes or steps left before merging this PR?
And how to proceed how to proceed with the merging process? (is it by adding `bors merge` comment to this PR?)

Comment (korthout) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java]: Thanks for the changes @ce-dmelnych 💯 

👍 LGTM 

💭 We've been moving away from merging with bors to using GitHub Merge Queues (see https://github.com/camunda/zeebe/issues/12097). You can simply click the `Merge when ready` button below to queue your pull request for merging.

Comment (ce-dmelnych) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java]: Thank you, @korthout!
Probably I'm lacking some permissions.

This is what I see in the PR merge section:

image

Comment (korthout) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java]: @abbasadel Can you give @ce-dmelnych access to the repo, for example by add @ce-dmelnych to `@camunda/zeebe-process-automation`?

Comment (abbasadel) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java]: ✅ Thanks @korthout.

Comment (ce-dmelnych) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java]: This issue was resolved!
Thank you,  @abbasadel for supporting me with it.

Comment (korthout) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java]: @ce-dmelnych Feel free to merge 👍

**Pull request merged**